test/e2e/data/infrastructure-vsphere/cluster-template-storage-policy.yaml

```diff
@@ -0,0 +1,209 @@
+apiVersion: v1
```

Comment:
is there a specific reason to have this file instead of using `packaging/flavorgen` ?

Reply:
Not really, was a pattern when working with CAPA e2e. Just needed a 'flavour' without the VsphereMachineTemplate. Should I just use `MultiNodeTemplateWithHAProxy` and then take out machine template?

Reply:
is there a reason why we don't need the `VSphereMachineTemplate` ?

Reply:
We definitely need it but the storage-policy name is determined at run time so the vsphereMachineTemplate is applied by the test once the storage-policy is set up.

Reply:
Ideally we should just consume a storage policy that is already created in our environment, this way we can re-use the same one across jobs, without having to generate it at runtime and deal with the naming issues.

Reply:
ya could do that. Went with creating and cleaning up per test so the user would have less setup to do. And the validation becomes simpler when it's a new storage policy rather than an existing one (can just make sure at least 1 vm is using it rather than checking against the vm-id is in the list of objects using the policy).

With using an existing sp, we'd still need a new flavour that specifies the storage-policy in the machine template. Or do you think I should just modify the quick start to use storage policy as well? We should cover both scenarios with tests imo.

Reply:
I prefer the programmatic way of doing the setup for creating a storage policy, which makes sure that there is no extra setup that needs to be done before running the test.

Reply:
I'd rather add changes to `flavorgen` to generate both (sp-only, and sp with datastore) and use an existing storage policy. which would allow to get rid of the file + test against an existing storage policy

Reply:
removed file and added storage policy to haproxy flavour.